public/app/plugins/datasource/azuremonitor/components/VariableEditor/VariableEditor.test.tsx

```diff
@@ -40,7 +40,16 @@ const defaultProps = {
   datasource: createMockDatasource({
     getSubscriptions: jest.fn().mockResolvedValue([{ text: 'Primary Subscription', value: 'sub' }]),
     getResourceGroups: jest.fn().mockResolvedValue([{ text: 'rg', value: 'rg' }]),
-    getMetricNamespaces: jest.fn().mockResolvedValue([{ text: 'foo/bar', value: 'foo/bar' }]),
+    getMetricNamespaces: jest
+      .fn()
+      .mockImplementation(
+        async (subscriptionId: string, resourceGroup?: string, resourceUri?: string, custom?: boolean) => {
+          if (custom !== true) {
+            return Promise.resolve([{ text: 'foo/bar', value: 'foo/bar' }]);
```

Comment:
I think the `Promise.resolve` is redundant here, as an `async` function that `return`s rather than `throw`s  just returns a resolved promise.

Reply:
Thanks for catching that! Removed it 😊